### PR TITLE
Adds --json flag to output JSON to STDOUT

### LINKED ISSUE #40: Output report as JSON or TOML or something else easily parsable
It'd be great to be able to execute something like 

    cargo audit --format json
    cargo audit --json

and get the report in a convenient format:

```json
{
  "database": "https://github.com/RustSec/advisory-db.git",
  "database-count": "10",
  "dependencies-count": "215",
  "vulnerabilities-found": true,
  "vulnerabilities": [
    {
      "ID": "RUSTSEC-2018-0003",
      "Crate": "smallvec",
      "Version": "0.2.1",
      "Date": "2018-07-19",
      "URL": "https://github.com/servo/rust-smallvec/issues/96",
      "Title": "Possible double free during unwinding in SmallVec::insert_many",
      "Solution": "upgrade to: >= 0.6.3 OR ^0.3.4 OR ^0.4.5 OR ^0.5.1"
    },
    {
      "ID": "RUSTSEC-2018-0003",
      "Crate": "smallvec",
      "Version": "0.2.1",
      "Date": "2018-07-19",
      "URL": "https://github.com/servo/rust-smallvec/issues/96",
      "Title": "Possible double free during unwinding in SmallVec::insert_many",
      "Solution": "upgrade to: >= 0.6.3 OR ^0.3.4 OR ^0.4.5 OR ^0.5.1"
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by colindean**
Fixes #40 

⚠️ This cannot be merged until 

- [X] rustsec/rustsec-crate#41 is merged 
    - [x] https://github.com/RustSec/cargo-audit/issues/32 is merged 
- [x] rustsec 0.9.3 is released.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,5 +18,7 @@ gumdrop_derive = "0.4"
 isatty = "0.1"
 lazy_static = "1"
 platforms = "0.1"
-rustsec = "0.9"
+rustsec = "0.9.3"
```
Comment (colindean) [inline Cargo.toml]: @tarcieri I realized just moments after I changed this to 0.9.3 that maybe it should be less specific. What should it be?

…Or you could merge it and change it to be less specific in another commit…

**Comment by tarcieri** [comment]
@colindean as it were I just released `0.9.3`, so that's fine...

**Comment by colindean** [comment]
It looks like this is good to go!

**Comment by tarcieri** [comment]
Great, thanks! And sorry it took so long

**Comment by colindean** [comment]
🎉
